esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -889,6 +889,51 @@ void EsdlServiceImpl::handleFinalRequest(IEspContext &context,
                    "No target URL configured for %s!", mthdef.queryMethodName());
     }
 
+    try
+    {
+        Owned<IPTree> responseTree = createPTreeFromXMLString(out, ipt_ordered);
+        auto responseResult = responseTree->queryBranch("//Results/Result");
+
+        if (responseResult)
+        {
+            auto replaceTag = [](StringBuffer& xml, const char* tag, const char* var)
+            {
+                VStringBuffer replaceThis("<%s>", tag);
+                VStringBuffer withThat("<%s name=\"%s\">", var, tag);
+
+                xml.replaceString(replaceThis, withThat);
+
+                replaceThis.insert(1, '/');
+                withThat.setLength(strlen(var) + 1);
+                withThat.insert(1, '/');
+                withThat.append('>');
+
+                xml.replaceString(replaceThis, withThat);
+            };
+
+            if (isroxie && tgtcfg->hasProp("@queryname"))
+                replaceTag(soapmsg, tgtcfg->queryProp("@queryname"), "roxiequery");
+            replaceTag(soapmsg, mthdef.queryRequestType(), "esprequest");
+            soapmsg.replaceString("xsdl:", "xsdl_");
+
+            Owned<IPTree> requestTree = createPTreeFromXMLString(soapmsg, ipt_ordered);
```

Comment:
You could probably do all of this more efficiently with a pull parser.  Even replace the tags that need replacing and copy accross the body tree into the out stringbuffer directly.

Unless we think the fully transformed requests will never be big enough for the efficiency savings to matter.